File: Count Subarrays With Fixed Bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
         long result = 0;

        int min = -1;
        int max = -1;
        int bad = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == minK) {
                min = i;
            } else if (nums[i] < minK) {
                min = -1;
                max = -1;
                bad = i+1;
            }
            if (nums[i] == maxK) {
                max = i;
            } else if (nums[i] > maxK) {
                max = -1;
                min = -1;
                bad = i+1;
            }

            if (min >= 0 && max >= 0) {
                result += Math.min(max, min) - bad + 1;
            }

        }
        return result;
    }
}

File: Count Subarrays With Fixed Bounds.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, int minK, int maxK) {

        long res = 0;  //for storing the ans

        int start = 0;  //storing the index after we get the maxinum element
        int minStart = 0 , maxStart = 0;  //store the min and max Index of element 

        bool minF = false , maxF = false;   //check for minimum and maximun element 

        for(int i = 0; i<nums.size();i++)
        {
            int num = nums[i];

            if(num < minK || num > maxK)   //edge case
            {
                minF = false ;
                maxF = false ; 
                start = i+1;   //store the index after the max element
            }
            if(num == minK)   // if element is equal to min
            {
                minF = true;  //mark as minimum as true
                minStart = i;   //store its index
            }  
            if(num == maxK)      //if element is equal to max element
           {
                maxF = true;  //mark as true
                maxStart = i;   //store its index
            }

            //cout<<res<<" ";
            // cout<<maxStart << " "<<endl;
            // cout<<minStart << " ";

            if(minF && maxF)         //if we find both min and max index as true
            {
                //count the element between the min and max element
                res += (min(minStart,maxStart) - start + 1);
            }
        }
        return res;
        
    }
};
